### ISSUE TITLE
CollateX format

### ISSUE THREAD

**Issue by tarrinw** [tla/stemmatology#43]
I realise that there hasn't been much development on this in the last decade, but I'm still interested as a perl user of 25 years and working on the text editing and analysis tool at menotag.ku.dk

I'm putting in CollateX JSON output but running into two problems: CollateX may have changed its format since this tool was updated, because (when using the JSON string):

Can't use an undefined value as an ARRAY reference at /usr/local/share/perl5/Text/Tradition/Parser/JSON.pm line 160, <DATA> line 41.

Additionally, I'm using string input because Text::Tradition won't read the JSON file:

malformed JSON string, neither array, object, number, string or atom, at character offset 0 (before "(end of string)") at /usr/local/share/perl5/Text/Tradition/Parser/JSON.pm line 117.

Where the JSON file is perfectly fine.

**Comment by tla** [comment]
Indeed, I haven't looked at this code in a very long time! The "malformed JSON" error suggests that it is trying to read an empty file (if character 0 is just before "(end of string)"); I would double check that you are giving the right filename as input.

If you are willing to branch out from Perl, the successor to this software is at https://github.com/DHUniWien/tradition_repo/; that is a Neo4J-backed Java Jersey repository that has the same functionality.
